### PR TITLE
New view model (latest one)

### PR DESCRIPTION
**Pull request by srsingh04**
I don't even know if this will work...

### CODE REVIEW

**Review by alejandrocalles — CHANGES_REQUESTED** [review]
I found a major issue: If you try to add a tag and you remove it, you can't add it again.

This might be related to the `onDismiss` function in `InputChipFun`. This function should be one that removes the tag from the tag list. For example. you could call it like this:
`InputChipFun(tag.name) { tagList.remove(tag) }`. I think will not work with the way the `tagList` is initialized here, so it might need some tweaking.

Apart from that, everything looks good. Thanks a lot for working on this! I left a couple of comments, but they're not that important.

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+import com.github.swent.echo.data.model.Semester
+import com.github.swent.echo.data.model.SemesterEPFL
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.ui.navigation.NavigationActions
+import com.github.swent.echo.ui.navigation.Routes
+import com.github.swent.echo.viewmodels.authentication.CreateProfileViewModel
+import com.github.swent.echo.viewmodels.tag.TagViewModel
+
+@Composable
+fun ProfileCreationScreen(
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:72]: see  #206 to follow up on this

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+ * A composable function that displays the UI for creating a user profile.
+ *
+ * @param sectionList The list of sections to be displayed in the dropdown menu.
+ * @param semList The list of semesters to be displayed in the dropdown menu.
+ * @param tagList The list of tags to be displayed as chips.
+ */
+@OptIn(ExperimentalLayoutApi::class)
+@SuppressLint("RestrictedApi")
+@Composable
+fun ProfileCreationUI(
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:108]: see  #206 to follow up on this

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                ) {
+                    Icon(Icons.Default.Add, "Add tags")
+                }
+            }
+            Spacer(modifier = Modifier.weight(1f))
+            // Save button
+            OutlinedButton(
+                onClick = {
+                    if (firstName.isBlank() || lastName.isBlank()) {
+                        showError = true
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:192]: see #207 to continue this later

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+                },
+                modifier = Modifier.fillMaxWidth().testTag("Save")
+            ) {
+                Text(text = stringResource(id = R.string.profile_creation_save_button))
+            }
+            if (showError) {
+                Snackbar(
+                    modifier = Modifier.padding(16.dp),
+                    action = {
+                        Button(onClick = { showError = false }) {
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:207]: also #207

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun InputChipFun(
+    text: String,
+    onDismiss: () -> Unit,
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:271]: see #208 for follow up

File: app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+) {
+    var enabled1 by remember { mutableStateOf(true) }
```
[the remark this answers is not in the document]
Comment (violoncelloCH) [reply app/src/main/java/com/github/swent/echo/compose/authentication/CreateProfile.kt:273]: follow up in #209

**Review by violoncelloCH — APPROVED** [review]
merging this as is due to time constraints as discussed
I split the remaining points to address into the following issues so they can be addressed one by one: #206 #207 #208 #209

**Pull request merged**